tiatoolbox/utils/misc_utils.py

```diff
@@ -0,0 +1,48 @@
+"""
+This file contains miscellaneous small functions repeatedly required and used in the repo
+"""
+import os
+import pathlib
+
+
+def split_path_name_ext(full_path):
+    """
+    Split path of a file to directory path, file name and extension
+
+    Args:
+        full_path: Path to a file
+
+    Returns:
+        input_dir: directory path
+        file_name: name of the file without extension
+        ext: file extension
+    """
+    input_dir, file_name = os.path.split(full_path)
+    file_name, ext = os.path.splitext(file_name)
+    return input_dir, file_name, ext
+
+
+def grab_files_from_dir(input_path, file_types=("*.jpg", "*.png", "*.tif")):
+    """
+    Grabs file paths specified by file extensions
+
+    Args:
+        input_path: path to the directory where files need to be searched
```

Comment:
Updated in this commit https://github.com/TIA-Lab/tiatoolbox/commit/66dac0f02e1fb7b3d349c1b5239f2583b4bde614. Please let me know if I have added this incorrectly otherwise resolve this conversation.